vtl-spark/src/main/java/fr/insee/vtl/spark/SparkAggrFuncExprBuilder.java

```diff
@@ -0,0 +1,90 @@
+package fr.insee.vtl.spark;
+
+import org.apache.spark.sql.Column;
+
+import java.util.*;
+
+import static org.apache.spark.sql.functions.*;
+
+public class SparkAggrFuncExprBuilder {
+
+    private final List<AbstractMap.SimpleImmutableEntry<String, String>> operations;
+    private final List<AbstractMap.SimpleImmutableEntry<String, String>> aliases;
+    private List<Column> tailExpressions;
+    private Column headExpression;
+    private final int accuracy = 1000000;
+
+    public SparkAggrFuncExprBuilder(List<AbstractMap.SimpleImmutableEntry<String, String>> operations,
+                                    List<AbstractMap.SimpleImmutableEntry<String, String>> aliases) throws Exception {
+        this.operations = operations;
+        this.aliases = aliases;
+        this.parseExpressions();
+    }
+
+    public List<Column> getTailExpressions() {
+        return tailExpressions;
+    }
+
+    public Column getHeadExpression() {
+        return headExpression;
+    }
+
+    private void parseExpressions() throws Exception {
+        tailExpressions = new ArrayList<>();
+        // get head expression
+        headExpression = buildExpression(operations.get(0).getKey(), operations.get(0).getValue(), aliases.get(0).getValue());
+
+        // get the tail expression
+        int index = 1;
+        while (index < operations.size()) {
+            tailExpressions.add(buildExpression(operations.get(index).getKey(), operations.get(index).getValue(), aliases.get(index).getValue()));
+            index++;
+        }
+
+
+    }
+
+    private Column buildExpression(String colName, String action, String aliasColName) throws Exception {
```

Comment:
We should throw a VtlScriptException here, we can discuss this next meeting.

Reply:
Agree, I'm just not familiar with  VTL Exception. So I throw a general execption.